### ISSUE TITLE
Blueprints together with through associations

### ISSUE THREAD

**Issue by Bazze** [balderdashy/sails#3096]
I'm having a issue where the junction table names are getting too long (> 64 chars) for MySQL to handle and I'm therefore trying to find the best way to alter those names. It has come to my attention that you could create your own junction table by using the "through" property and set the table name there, however, this seems to break the API blueprints when it comes to creating associated records.

Are the blueprints supposed to work as before, seamlessly, when you specify a "through" association? And also, could you point me to where in the code these automatically created junction table names are created? I'm not even sure in which package they are created. sails? waterline? sails-mysql? waterline-sequel?

**Comment by sailsbot** [comment]
Thanks for posting, @Bazze.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by Bazze** [comment]
I would still love some guidance on this. It's still a massive headache for me.
